src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -49,6 +63,49 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
 
   let nextUnitOfWork : ?Fiber = null;
 
+  let currentRootsWithPendingWork : ?Fiber = null;
+
+  function findNextUnitOfWork(priorityLevel : PriorityLevel) : ?Fiber {
+    let current = currentRootsWithPendingWork;
+    while (current) {
+      if (current.pendingWorkPriority !== 0 &&
+          current.pendingWorkPriority <= priorityLevel) {
+        // This node has work to do that fits our priority level criteria.
+        if (current.pendingProps !== null) {
+          // We found some work to do. We need to return the "work in progress"
+          // of this node which will be the alternate.
+          return current.alternate;
+        }
+        // If we have a child let's see if any of our children has work to do.
+        // Only bother doing this at all if the current priority level matches
+        // because it is the highest priority for the whole subtree.
+        if (current.child) {
+          current = current.child;
+          continue;
+        }
+        // If we match the priority but has no child and no work to do,
+        // then we can safely reset the flag.
+        current.pendingWorkPriority = NoWork;
```

Comment:
To be honest, I didn't really understand this well enough when I wrote it to structure it well from the beginning. The current organization of beginWork/completeWork/pendingWork/scheduler could use some restructuring for clarity but there are still unsolved problems so I'll do that when those are solved.

This isn't actually needed right now. Probably just something I left behind. I suspect that with setState and more edge cases it might be come needed.

It's not the primary way of resetting priority.